docs: add documentation for policy-cidr-match-mode=nodes (#28421)
cilium-team/cilium: Documentation/security/policy/language.rst

        .. literalinclude:: ../../../examples/policies/l3/cidr/cidr.json

.. _cidr_select_nodes:

Selecting nodes with CIDR / ipBlock
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. include:: ../../beta.rst

By default, CIDR-based selectors do not match in-cluster entities (pods or nodes).

> Optionally, the policy engine can be directed to select nodes via CIDR / ipBlock.

Reviewer: No Latin abbreviations
```suggestion
Optionally, the policy engine can be directed to select nodes by CIDR / ipBlock.
```

Author: But how else am I going to show off my overpriced education :stuck_out_tongue_closed_eyes:

This requires Cilium to be configured with ``--policy-cidr-match-mode=nodes`` or
the equivalent Helm value ``policyCIDRMatchMode: nodes``. It is safe to toggle this

> option on a running cluster, nor does it affect upgrades or downgrades.

Reviewer: Please clarify what you mean by "it" here: toggling? If so:
```suggestion
option on a running cluster, nor does toggling the option affect upgrades or downgrades.
```

> When this is enabled, every node will have a distinct local :ref:`security identity <security_identities>`
> allocated for it, which will cause a minor increase in memory usage on very large clusters.

Reviewer: Specific is better than general, use present tense
```suggestion
When CIDR/ipblock node selection is enabled, every node has a distinct local :ref:`security identity <security_identities>` allocated for it, which causes a minor increase in memory usage on very large clusters.
```

Author: reworded it to be more precise anyways, ptal

        .. literalinclude:: ../../../examples/policies/l3/cidr/cidr.json

.. _cidr_select_nodes:

Selecting nodes with CIDR / ipBlock
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. include:: ../../beta.rst

By default, CIDR-based selectors do not match in-cluster entities (pods or nodes).

> Optionally, the policy engine can be directed to select nodes by CIDR / ipBlock.
> This requires Cilium to be configured with ``--policy-cidr-match-mode=nodes`` or

Reviewer: Use active voice, address your reader as "you"
```suggestion
Optionally, you can direct the policy engine to select nodes by CIDR / ipBlock.
This requires you to configure Cilium with ``--policy-cidr-match-mode=nodes`` or
```

Reviewer: @squeed I'm actually going to approve this now because I'll be OOO next week and I don't want this PR held up by my own sensibilities. Feel free to open a follow-up PR if you like.

the equivalent Helm value ``policyCIDRMatchMode: nodes``. It is safe to toggle this
option on a running cluster, and toggling the option affects neither upgrades nor downgrades.

When this is enabled, every agent will allocate a distinct local

> :ref:`security identity <security_identities>` for all other nodes. This will cause

Reviewer 2: Minor nit: I see the security identities doc as a developer-focused doc around bitpatterns that is not an API and not user-facing, subject to change without warning. Of course we would only change it if we see a reason and if it's not likely to impact user traffic. I'm a little bit concerned that by cross-linking that doc from this network policy doc which is clearly user-facing and API, we might be giving the impression of stronger guarantees in that other doc than what we intend to.

I think that if we're trying to cross-link here for some specific reason, we should be linking to the user-facing docs like https://docs.cilium.io/en/stable/security/network/identity/#identity-based or similar.

> When this is enabled, every agent will allocate a distinct local
> :ref:`security identity <security_identities>` for all other nodes. This will cause
> a minor increase in memory usage on very large clusters.

Reviewer: Specific is better than general, use present tense
```suggestion
When CIDR/ipBlock node selection is enabled, every agent allocates a distinct local
:ref:`security identity <security_identities>` for all other nodes. This causes
a minor increase in memory usage on very large clusters.
```

> a minor increase in memory usage on very large clusters.

Reviewer 2: how large are we talking? Is it worth highlighting? Maybe take 5K nodes as a target, are we talking ~1MiB, 10MiB, 100MiB?

Author: @nathanjsweet did the actual benchmarking; what was the difference you saw? I can't recall.

(This line was added for box-checking.)

Reviewer 3: Call it 0.5-1kB per identity (it can vary). So 5-10MB for 10k nodes, ~50-100MB for 100k, etc.

Author: Updated, thanks!

        .. literalinclude:: ../../../examples/policies/l3/cidr/cidr.json

.. _cidr_select_nodes:

Selecting nodes with CIDR / ipBlock
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. include:: ../../beta.rst

By default, CIDR-based selectors do not match in-cluster entities (pods or nodes).
Optionally, you can direct the policy engine to select nodes by CIDR / ipBlock.
This requires you to configure Cilium with ``--policy-cidr-match-mode=nodes`` or
the equivalent Helm value ``policyCIDRMatchMode: nodes``. It is safe to toggle this
option on a running cluster, and toggling the option affects neither upgrades nor downgrades.

When ``--policy-cidr-match-mode=nodes`` is specified, every agent allocates a
distinct local :ref:`security identity <arch_id_security>` for all other nodes.
This causes a minor increase in memory usage on very large clusters.

> This is particularly relevant to self-hosted clusters -- that is, clusters where

Author: @nathanjsweet / @zacharysarah want to review this paragraph?

Reviewer 3: Thanks! LGTM.
